src/Sarif.Viewer.VisualStudio/Tags/SarifLocationErrorTagger.cs

```diff
@@ -0,0 +1,139 @@
+﻿// Copyright (c) Microsoft. All rights reserved. 
+// Licensed under the MIT license. See LICENSE file in the project root for full license information. 
+
+namespace Microsoft.Sarif.Viewer.Tags
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using Microsoft.Sarif.Viewer.ErrorList;
+    using Microsoft.VisualStudio.Shell;
+    using Microsoft.VisualStudio.Text;
+    using Microsoft.VisualStudio.Text.Tagging;
+
+    /// <summary>
+    /// Provides tags to Visual Studio from <see cref="SarifErrorListItem"/>s.
+    /// </summary>
+    /// <remarks>
+    /// The tags provided from this class represent all the instances of <see cref="ResultTextMarker"/> that a <see cref="SarifErrorListItem"/> may contain
+    /// that are children of the currently selected <see cref="SarifErrorListItem"/> as defined by <see cref="ISarifErrorListEventSelectionService.SelectedItem"/>.
+    /// This allows us to provide error tags for code flows for a specific result without displaying potentially overlapping code flow error tags from other results.
```

Comment:
>results [](start = 157, length = 7)

For a given SarifErrorListItem, this class provides one squiggle for the error location and multiple tool tips for all locations related to this result (the error itself, related locations, code flows, etc.).